EIPS/eip-5593.md

```diff
@@ -0,0 +1,76 @@
+---
+eip: 5593
+title: Restrict Web3 Provider Object API Injection
+description: Wallet guidance for restricting web3 provider object API access to secure contexts for improved privacy and security for wallet users.
+author: Yan Zhu (@diracdeltas), Brian R. Bondy (@bbondy), Kyle Den Hartog (@kdenhartog)
+discussions-to:  https://ethereum-magicians.org/t/rfc-limiting-provider-object-injection-to-secure-contexts/10670
+status: Draft
+type: Standards Track
+category (*only required for Standards Track): Interface
+created: 2022-09-05
+requires (*optional): <EIP number(s)>
+---
+
+## Abstract
+
+Historically the web platform has had a notion of “powerful” APIs like [geolocation](https://www.w3.org/TR/geolocation/) and [camera/microphone](https://www.w3.org/TR/mediacapture-streams/), which are subject to additional security restrictions such as those defined by [secure contexts](https://www.w3.org/TR/secure-contexts/). Since the web3 provider object APIs allow dApp websites to request access to sensitive user data and to request use of user funds, new web3 provider object APIs generally should align to the security considerations of other powerful APIs like geolocation.
```

Comment:
No external links

Reply:
What's the best way to externally reference other normative specs and useful documentation then to enhance context for readers and implementers? Within W3C specs there's something similar to citation bibliography that they use. This would be possible to achieve in a similar method using markdowns [reference style links](https://www.markdownguide.org/basic-syntax/). 

This has the benefit of better readability while also being able to externally link for additional context. I assume there's a reason this rule was put in place though and this benefit was outweighed by the downsides of using them. 

Out of curiosity what's the motivation for this rule as? I assume either security related or to prevent maintaining broken links, but am not certain.

I'd be happy to help define some additional editorial rules in the process document to allow for external links if the editor team is open to revisiting this rule.

Reply:
@kdenhartog I think it's complete reasonable to include reference to w3c. The editors are still debating rules about external links. 

FYI: @lightclient @gcolvin for comments

Reply:
External links are allowed by EIP-1 when necessarily. 
Please see https://github.com/ethereum/EIPs/issues/5597

Reply:
The current policy is to disallow external links in their entirety. If the linked resources allows redistribution, you may embed it in `assets/eip-5593` and link to it from there.

In this case, I don't think the links are necessary to understand the proposal, so I'd just remove them.